### PR TITLE
Improvements and fixes for target point (GSNGM-706, GSNGM-722, GSNGM-698, GSNGM-712)

### PR DESCRIPTION
**Pull request by vladyslav-tk**

### CODE REVIEW

File: src/cesiumutils.ts [hunk 1]
```diff
@@ -392,24 +385,43 @@ export function getValueOrUndefined(prop) {
...
-  const rightCartographic = Cartographic.fromCartesian(right);
-  const leftUp = Cartesian3.fromRadians(rightCartographic.longitude, leftCartographic.latitude);
-  const rightDown = Cartesian3.fromRadians(rightCartographic.longitude, rightCartographic.latitude);
-  const isEast = rightCartographic.longitude > leftCartographic.longitude;
-  const isNorth = rightCartographic.latitude > leftCartographic.latitude;
-  return new Cartesian3(
-    Cartesian3.distance(left, leftUp) * (isEast ? 1 : -1),
-    Cartesian3.distance(leftUp, rightDown) * (isNorth ? 1 : -1),
-    rightCartographic.height - leftCartographic.height
+export function lookAtPoint(position: Cartesian3, camera: Camera) {
```
Comment (gberaudo) [inline src/cesiumutils.ts:392]: Shouldn't we use https://cesium.com/learn/cesiumjs/ref-doc/Camera.html#lookAt instead?

File: src/elements/ngm-height-slider.ts [hunk 2]
```diff
@@ -62,12 +62,15 @@ export class NgmHeightSlider extends LitElementI18n {
     }
   }
 
+  //
+// @pointerdown=${() => (<NgmNavTools>document.querySelector('ngm-nav-tools')).stopTracking()}
+// @pointerup=${() => (<NgmNavTools>document.querySelector('ngm-nav-tools')).continueTracking()}
+
```
Comment (gberaudo) [inline src/elements/ngm-height-slider.ts:68]: Unused

File: src/elements/ngm-nadir-view.ts [hunk 3]
```diff
@@ -64,20 +69,28 @@ export class NgmNadirView extends LitElementI18n {
...
       this.pitch = camera.pitch;
       this.heading = camera.heading;
+      const targetPoint = getTargetParam();
+      if (targetPoint) {
+        this.position = Cartesian3.clone(camera.positionWC, new Cartesian3());
+        updateHeightForCartesianPositions([targetPoint], camera.positionCartographic.height, undefined, true);
+      }
       this.viewer.camera.setView({
+        destination: targetPoint,
+        convert: true,
```
Comment (gberaudo) [inline src/elements/ngm-nadir-view.ts:93]: Documentation says we don't need this attribute:  optional Whether to convert the destination from world coordinates to scene coordinates (only relevant when not using 3D). Defaults to true.

File: src/elements/ngm-nav-tools.ts [hunk 4]
```diff
@@ -159,21 +170,35 @@ export class NgmNavTools extends LitElementI18n {
...
+    this.viewer!.scene.camera.lookAtTransform(Matrix4.IDENTITY);
+  }
+
+  startTracking() {
+    this.addTargetPoint(this.refIcon.position!.getValue(this.julianDate));
+    const center = this.refIcon.position!.getValue(this.julianDate);
+    const camera = this.viewer!.camera;
+    lookAtPoint(center, camera);
+    const transform = Transforms.eastNorthUpToFixedFrame(center);
+    camera.lookAtTransform(transform);
```
Comment (gberaudo) [inline src/elements/ngm-nav-tools.ts:196]: Why do a lookAtPoint first then a lookAtTransform? Is the first line useful?

**Review by gberaudo — APPROVED** [review]
See my comments, otherwise LGTM.

File: src/cesiumutils.ts (hunk 1 above)
Comment (vladyslav-tk) [reply src/cesiumutils.ts:392]: no, it works differently

File: src/elements/ngm-nav-tools.ts (hunk 4 above)
Comment (vladyslav-tk) [reply src/elements/ngm-nav-tools.ts:196]: lookAtTransform start tracking of position

File: src/elements/ngm-height-slider.ts (hunk 2 above)
Comment (vladyslav-tk) [reply src/elements/ngm-height-slider.ts]: forgot to remove, thanks

File: src/elements/ngm-nadir-view.ts (hunk 3 above)
Comment (vladyslav-tk) [reply src/elements/ngm-nadir-view.ts]: same here, thanks

**Pull request merged**